### PR TITLE
chore: upgrade maplibre-gl-js-amplify to v2.1.0 to support getSuggestions and getPlace

### PR DESCRIPTION
**Pull request by sreeramsama**
#### Description of changes
Upgrade maplibre-gl-js-amplify to v2.1.0 to support getSuggestions and getPlace
Docs PR: https://github.com/aws-amplify/docs/pull/4615

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by wlee221 — DISMISSED** [review]

File: packages/react/package.json [hunk 1]
```diff
@@ -67,7 +67,7 @@
     "lodash": "4.17.21",
     "mapbox-gl": "1.13.1",
     "maplibre-gl": "2.1.9",
-    "maplibre-gl-js-amplify": "2.0.4",
+    "maplibre-gl-js-amplify": "2.1.0",
```
Comment (wlee221) [inline packages/react/package.json:70]: nit: we should still make a changeset because this changes customer surface area

Comment (sreeramsama) [reply packages/react/package.json:70]: Done. Added changeset.

**Review by wlee221 — DISMISSED** [review]

**Review by thaddmt — APPROVED** [review]

**Review by thaddmt — DISMISSED** [review]

File: packages/e2e/cypress/integration/common/geo.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ When('I clear the search results', () => {
   cy.findByRole('textbox', {
     name: /search/i,
   }).trigger('mouseenter');
-  cy.findByRole('button', { name: 'Clear' }).click();
+  cy.findByRole('button', { name: 'Clear' }).click({ force: true });
```
Comment (wlee221) [inline packages/e2e/cypress/integration/common/geo.ts:19]: Why is this necessary? Can we add a comment if this is not immediately resolvable?

Comment (sreeramsama) [reply packages/e2e/cypress/integration/common/geo.ts:19]: 'Clear' button somehow couldn’t be found by cypress when its display property is being toggled, so added {force: true} for the click. I see it is already being used in other places so thought maybe it is an existing Cypress issue.

**Review by thaddmt — APPROVED** [review]

File: packages/e2e/cypress/integration/common/geo.ts (hunk 2 above)
Comment (wlee221) [reply packages/e2e/cypress/integration/common/geo.ts:19]: Understood -- ideally, we should avoid `force: true` if possible, or explain in comment why this is necessary. For example: https://github.com/aws-amplify/amplify-ui/blob/c11de75eaf0e0476cea56d7cd2da1ffa04d1c639/packages/e2e/cypress/integration/common/shared.ts#L178-L184

Not blocking, but I think it's worth looking at for test maintainability purposes.

**Review by thaddmt — APPROVED** [review]

**Review by wlee221 — DISMISSED** [review]

**Review by thaddmt — DISMISSED** [review]

File: packages/e2e/cypress/integration/common/geo.ts [hunk 3]
```diff
@@ -16,7 +16,9 @@ When('I clear the search results', () => {
   cy.findByRole('textbox', {
     name: /search/i,
   }).trigger('mouseenter');
-  cy.findByRole('button', { name: 'Clear' }).click();
+  // Adding 'force' as the clear button is hidden until we hover on textbox,
+  // and the click action seems to happen before the hover thus failing the check for the clear button element.
```
Comment (wlee221) [inline packages/e2e/cypress/integration/common/geo.ts:20]: Last nit, I promise 🙏 :

```suggestion
  /** 
  /**
   * Adding 'force' as the clear button is hidden until we hover on textbox,
   * and the click action happens before the hover.
   */
```

Comment (sreeramsama) [reply packages/e2e/cypress/integration/common/geo.ts]: Updated

**Review by thaddmt — APPROVED** [review]

**Review by wlee221 — APPROVED** [review]
